substrate/frame/election-provider-support/src/lib.rs

```diff
@@ -674,10 +746,149 @@ pub type Voter<AccountId, Bound> = (AccountId, VoteWeight, BoundedVec<AccountId,
+	fn clone(&self) -> Self {
+		Self { voters: self.voters.clone(), total: self.total }
+	}
+}
+
+impl<AccountId, Bound: Get<u32>> TryFrom<sp_npos_elections::Support<AccountId>>
+	for BoundedSupport<AccountId, Bound>
+{
+	type Error = &'static str;
+	fn try_from(s: sp_npos_elections::Support<AccountId>) -> Result<Self, Self::Error> {
+		let voters = s.voters.try_into().map_err(|_| "voters bound not respected")?;
+		Ok(Self { voters, total: s.total })
+	}
+}
+
+/// A bounded vector of [`BoundedSupport`].
+///
+/// A [`BoundedSupports`] is a set of [`sp_npos_elections::Supports`] which are bounded in two
+/// dimensions. `BInner` corresponds to the bound of the maximum backers per voter and `BOuter`
+/// corresponds to the bound of the maximum winners that the bounded supports may contain.
+///
+/// With the bounds, we control the maximum size of a bounded supports instance.
+#[derive(Encode, Decode, TypeInfo, DefaultNoBound, MaxEncodedLen)]
+#[codec(mel_bound(AccountId: MaxEncodedLen, BOuter: Get<u32>, BInner: Get<u32>))]
+#[scale_info(skip_type_params(BOuter, BInner))]
+pub struct BoundedSupports<AccountId, BOuter: Get<u32>, BInner: Get<u32>>(
+	pub BoundedVec<(AccountId, BoundedSupport<AccountId, BInner>), BOuter>,
+);
+
+impl<AccountId: Debug, BOuter: Get<u32>, BInner: Get<u32>> Debug
+	for BoundedSupports<AccountId, BOuter, BInner>
+{
+	fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+		for s in self.0.iter() {
+			write!(f, "({:?}, {:?}, {:?}) ", s.0, s.1.total, s.1.voters)?;
+		}
+		Ok(())
+	}
+}
+
+impl<AccountId: PartialEq, BOuter: Get<u32>, BInner: Get<u32>> PartialEq
+	for BoundedSupports<AccountId, BOuter, BInner>
+{
+	fn eq(&self, other: &Self) -> bool {
+		self.0 == other.0
+	}
+}
+
+impl<AccountId, BOuter: Get<u32>, BInner: Get<u32>>
+	From<BoundedVec<(AccountId, BoundedSupport<AccountId, BInner>), BOuter>>
+	for BoundedSupports<AccountId, BOuter, BInner>
+{
+	fn from(t: BoundedVec<(AccountId, BoundedSupport<AccountId, BInner>), BOuter>) -> Self {
+		Self(t)
+	}
+}
+
+impl<AccountId: Clone, BOuter: Get<u32>, BInner: Get<u32>> Clone
+	for BoundedSupports<AccountId, BOuter, BInner>
+{
+	fn clone(&self) -> Self {
+		Self(self.0.clone())
+	}
+}
+
+impl<AccountId, BOuter: Get<u32>, BInner: Get<u32>> sp_std::ops::Deref
+	for BoundedSupports<AccountId, BOuter, BInner>
+{
+	type Target = BoundedVec<(AccountId, BoundedSupport<AccountId, BInner>), BOuter>;
+
+	fn deref(&self) -> &Self::Target {
+		&self.0
+	}
+}
+
+impl<AccountId, BOuter: Get<u32>, BInner: Get<u32>> IntoIterator
+	for BoundedSupports<AccountId, BOuter, BInner>
+{
+	type Item = (AccountId, BoundedSupport<AccountId, BInner>);
+	type IntoIter = sp_std::vec::IntoIter<Self::Item>;
+
+	fn into_iter(self) -> Self::IntoIter {
+		self.0.into_iter()
+	}
+}
+
+/// An extension trait to convert from [`sp_npos_elections::Supports<AccountId>`] into
+/// [`BoundedSupports`].
+pub trait TryIntoBoundedSupports<AccountId, BOuter: Get<u32>, BInner: Get<u32>> {
```

Comment:
Was this my code? I can't remember why it cannot be 

`TryInto<BoundedSupport<_, _, _>>`

Reply:
I will try to simplify and remove the need for this trait.

Reply:
I think probably because we need to somehow pass the bounds. But I still feel the trait is not very useful, since you should be able to build `BoundedSupports` only from `Supports`. 

I would remove the trait, and have a fn implemented directly on the type `Supports`.

Reply:
I removed this trait and implemented the `TryFrom<sp_npos_elections::Supports<AccountId>>` directly into `BoundedSupports`.